samples/nrf9160/azure_iot_hub/Kconfig

```
@@ -14,4 +14,7 @@ module = AZURE_IOT_HUB_SAMPLE
 module-str = Azure IoT Hub sample
 source "${ZEPHYR_BASE}/subsys/logging/Kconfig.template.log_config"
 
+config DEVICE_ID_USE_IMEI
+	bool "Use IMEI as device ID when connecting to Azure IOT hub"
```

Since HW info is only needed when this option is set we could simply select the needed configs.

```suggestion
	bool "Use IMEI as device ID when connecting to Azure IOT hub"
	select HW_ID_LIBRARY
        select HW_ID_LIBRARY_SOURCE_IMEI
```

---

HW_ID_LIBRARY_SOURCE_IMEI is a `choice` parameter, so it can't be set in a Kconfig file unfortunately.